includes/classes/DistributorPost.php

```diff
@@ -0,0 +1,154 @@
+	 *    @type WP_Term[] Post terms keyed by taxonomy.
+	 * }
+	 */
+	public $terms = [];
+
+	/**
+	 * Distributable post media.
+	 *
+	 * @var array[] Array of media objects.
+	 */
+	public $media = [];
+
+	/**
+	 * Whether this is the source (true) or a distributed post (false).
+	 *
+	 * @var bool
+	 */
+	public $is_source = true;
+
+	/**
+	 * Whether this post is linked to the original version.
+	 *
+	 * For the original post this is set to true.
+	 *
+	 * @var bool
+	 */
+	public $is_linked = true;
+
+	/**
+	 * The original post ID.
+	 *
+	 * @var int
+	 */
+	public $original_post_id = 0;
+
+	/**
+	 * The type of connection this post is distributed from.
+	 *
+	 * @var string internal|external|pushed|empty (for source)
+	 */
+	public $connection_type = '';
+
+	/**
+	 * The connection ID this post is distributed from.
+	 *
+	 * For internal connections this is the site ID. For external connections
+	 * this refers to the connection ID.
+	 *
+	 * @var int
+	 */
+	public $connection_id = 0;
+
+	/**
+	 * Initialize the DistributorPost object.
+	 *
+	 * @param WP_Post|int $post WordPress post object or post ID.
+	 */
+	public function __construct( $post ) {
+		$post = get_post( $post );
+
+		if ( ! $post ) {
+			return;
+		}
+
+		$this->post = $post;
+
+		// Set up the distributable data.
+		$this->meta  = Utils\prepare_meta( $post->ID );
+		$this->terms = Utils\prepare_taxonomy_terms( $post->ID );
+		$this->media = Utils\prepare_media( $post->ID );
+
+		/*
+		 * The original post ID is listed as excluded post meta and therefore
+		 * unavailable in the meta property. We need to get it using the post
+		 * meta API.
+		 */
+		$original_post_id = get_post_meta( $post->ID, 'dt_original_post_id', true );
+		if ( empty( $original_post_id ) ) {
+			// This is the source post.
+			$this->is_source = true;
+			$this->is_linked = true;
+			return;
+		}
+
+		// Set up information for a distributed post.
+		$this->is_source = false;
+		// Reverse value of the `dt_unlinked` meta data.
+		$this->is_linked = ! get_post_meta( $post->ID, 'dt_unlinked', true );
+
+		// Determine the connection type.
+		if ( get_post_meta( $post->ID, 'dt_original_blog_id', true ) ) {
+			/*
+			 * Internal connections store the original blog's ID.
+			 *
+			 * Pushed and pulled posts are indistinguishable from each other.
+			 */
+			$this->connection_type = 'internal';
+			$this->connection_id   = get_post_meta( $post->ID, 'dt_original_blog_id', true );
+		} elseif ( get_post_meta( $post->ID, 'dt_original_source_id', true ) ) {
+			// Post was pulled from an external connection.
+			$this->connection_type = 'external';
+			$this->connection_id   = get_post_meta( $post->ID, 'dt_original_source_id', true );
+		} elseif ( get_post_meta( $post->ID, 'dt_subscription_signature', true ) ) {
+			// Post was pushed from an external connection.
+			$this->connection_type = 'pushed';
+			$this->connection_id   = get_post_meta( $post->ID, 'dt_subscription_signature', true );
+		}
```

Comment:
This is broken...

`dt_original_source_id` is included in both pushed and pulled posts.

In pushed posts, `dt_original_source_id` is stored in the distributed copy as the connection ID on the source server. IE, the distributed copy does not have access to the post data.

In pulled posts, `dt_original_source_id` is stored in the distrubuted copy as the connection on on the source server, IE the distributed copy does have access to the post data.